// @britishcouncil/react-solas/dist/components/alerts/Alert.js
import React from 'react';
import cn from 'classnames';
export const Alert = props => {
    const onClose = () => {
        props.onClose && props.onClose();
    };
    const classes = cn(`alert alert-${props.type}`, props.className, {
        'alert-dismissible': props.dismissible,
    });
    return (React.createElement("div", {
            role: "alert",
            ...props,
            className: classes
        },
        props.dismissible ? React.createElement(DismissButton, {
            onClose: onClose
        }) : React.createElement(React.Fragment, null),
        props.children || React.createElement(React.Fragment, null)));
};
const DismissButton = (props) => {
    return (React.createElement("button", {
            type: "button",
            className: "close",
            "data-dismiss": "alert",
            "aria-label": "Close",
            onClick: props.onClose
        },
        React.createElement("span", {
            "aria-hidden": "true"
        }, "\u00D7")));
};
