// @britishcouncil/react-extro-ui/dist/components/fileUploader/FileUploader.style.js
import styled from '@emotion/styled';
export const Container = styled.div `
  cursor: pointer;
  border: 2px dashed ${({ theme }) => theme.colors.lightGrey};
  height: 300px;
  border-radius: 12px;
  color: ${({ theme }) => theme.colors.darkGrey};
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  &:active {
    background: #eaf4ff;
    border-color: ${({ theme }) => theme.colors.darkBlue};
  }

  &.on-drag {
    background: #eaf4ff;
    border-color: ${({ theme }) => theme.colors.darkBlue};
  }

  &.disabled {
    pointer-events: none;
    opacity: 0.5;
  }
`;
export const Title = styled.h3 `
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 30px;
  color: ${({ theme }) => theme.colors.indigo});
  margin-bottom: 40px;
`;
export const Text = styled.p `
  font-size: 18px;
  margin-top: 24px;

  span {
    color: ${({ theme }) => theme.colors.darkBlue};
    text-decoration: underline;
  }
`;
export const SmallText = styled.p `
  font-size: 14px;
  margin-top: 16px;
`;
export const FilesList = styled.div `
  opacity: ${({ loading }) => (loading ? 0.5 : 1)};
  display: flex;
  flex-direction: column;
  gap: 16px;
  margin-top: 16px;
`;
export const FileItem = styled.div `
  border: 1px solid ${({ theme }) => theme.colors.lightGrey};
  padding: 16px;
  border-radius: 12px;
  color: ${({ theme }) => theme.colors.darkGrey};
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 23px;

  span {
    cursor: pointer;
  }
`;
export const FileName = styled.div `
  display: flex;
  align-items: center;
  gap: 23px;
`;
export const Error = styled.p `
  margin: 32px 0;
  color: ${({ theme }) => theme.colors.darkRed};
`;
