// @britishcouncil/react-extro-ui/dist/components/table/components/pagination/Pagination.styles.js
import styled from '@emotion/styled';
import {
    Select
} from '../../../select';
export const Root = styled.div `
  margin-top: 2rem;
  display: flex;
  align-items: center;
  gap: 10%;
  justify-content: ${({ showAllResultsCount }) => (showAllResultsCount ? 'flex-start' : 'center')};
  button {
    cursor: pointer;
    &:focus-visible {
      box-shadow: ${({ theme: { colors } }) => `
0 0 0 0.4 rem $ {
    colors.orange
}, 0 0 0 0.4 rem $ {
    colors.orange
}
`};
      outline: none;
    }
    &:disabled {
      color: ${({ theme }) => theme.colors.grey};
      cursor: auto;
    }
  }
`;
export const AllResults = styled.div `
  display: flex;
  align-items: center;
  gap: 4px;
  border: ${({ theme }) => `
1 px solid $ {
    theme.colors.lightGrey
}
`};
  border-radius: 6px;
  padding: 8px;
  font-size: ${({ theme }) => `
1 px solid $ {
    theme.fontSize.desktop.small
}
`};
  background: white;
`;
export const PaginationRoot = styled.div `
  display: flex;
  justify-content: space-between;
  width: 60%;
  color: ${({ theme }) => theme.colors.darkBlue};

  /* Previous and Next arrows */
  & > button:first-of-type::before,
  & > button:last-of-type::after {
    content: '';
    display: inline-block;
    width: 10px;
    height: 10px;
    border-style: solid;
    border-color: inherit;
    transform: rotate(45deg);
    margin: 0 8px;
  }
  & > button:first-of-type::before {
    border-width: 0 0 2px 2px;
  }
  & > button:last-of-type::after {
    border-width: 2px 2px 0 0;
  }
`;
export const PageChooser = styled.div `
  display: flex;
  align-items: center;
  gap: 16px;
`;
export const Pages = styled.div `
  display: flex;
  gap: 16px;
  align-items: flex-start;

  button {
    text-decoration: underline;
  }
`;
export const Jump = styled.div `
  display: flex;
  align-items: center;
  gap: 8px;
  color: ${({ theme }) => theme.colors.darkGrey};
`;
export const StyledSelect = styled(Select)
`
  .extro-select__value-container {
    padding: 0 8px;
  }

  .extro-select__dropdown-indicator {
    padding: 0;
    margin-inline-end: 8px;
  }

  .extro-select__option {
    padding: 8px;
  }
`;
