// @britishcouncil/react-extro-ui/dist/components/stepper/Stepper.js
import React from 'react';
import * as S from './Stepper.style';
import {
    Step
} from './components';
import classnames from 'classnames';
export const Stepper = ({
    steps,
    activeStepIndex,
    testIdPrefix,
    showStepInformation = true
}) => {
    return (React.createElement(S.StepperContainer, {
            className: classnames({
                centered: !showStepInformation
            })
        },
        showStepInformation && activeStepIndex !== undefined && (React.createElement(S.StepInformations, null,
            React.createElement(S.StepName, null, steps[activeStepIndex]),
            React.createElement("p", null, `Step ${activeStepIndex + 1} of ${steps.length}`))),
        React.createElement(S.StepsContainer, {
            "data-testid": testIdPrefix ? `${testIdPrefix}-stepper` : undefined
        }, steps.map((_, index) => (React.createElement(React.Fragment, null,
            React.createElement(Step, {
                testId: testIdPrefix ? `${testIdPrefix}-stepper-step-${index}` : undefined,
                key: index,
                isActive: index === activeStepIndex,
                number: index + 1
            }),
            index < steps.length - 1 && React.createElement(S.Line, null)))))));
};
