// @britishcouncil/react-solas-ors3/dist/components/auth/AuthModal.js
import React, {
    useEffect,
    useState
} from 'react';
import AriaModal from '@justfixnyc/react-aria-modal';
import {
    LoadingContainer
} from '@britishcouncil/react-solas';
import styled from '@emotion/styled';
import {
    SuiPanel
} from '../layout';
import {
    AuthPanelBody
} from './AuthPanelBody';
import {
    LinkButton
} from '../general';
import {
    HeaderContainer,
    HeaderText
} from './styles';
export const AuthModal = ({
    show,
    onExit = () => null,
    loading,
    title = 'Complete your booking',
    closeLabel = 'Hide',
    onShow,
    onHide,
    ...props
}) => {
    const [prevState, setPrevState] = useState(null);
    useEffect(() => {
        if (prevState && !show && !!onHide) {
            onHide();
        }
        setPrevState(show);
    }, [show, onHide, prevState]);
    return show ? (React.createElement(AriaModal, {
            titleText: title,
            onExit: onExit,
            onEnter: onShow,
            verticallyCenter: true,
            initialFocus: "#register",
            underlayStyle: {
                paddingTop: '2em'
            }
        },
        React.createElement(SuiPanel, {
                intent: "default"
            },
            React.createElement(SuiPanel.Header, null,
                React.createElement(HeaderContainer, null,
                    React.createElement(HeaderText, null, title),
                    React.createElement(LinkButton, {
                        onClick: onExit,
                        "data-testid": "close"
                    }, closeLabel))),
            React.createElement(LoadingContainer, {
                    loading: loading
                },
                React.createElement(StyledLoginPanel, { ...props,
                    loading: loading
                }, props.children))))) : null;
};
const StyledLoginPanel = styled(AuthPanelBody)
`
  .bc-alert {
    max-width: 600px;
    border-radius: 0;
    margin: -26px -20px 20px;

    p {
      font-size: 0.8em;
      margin: 0 0 8px;
      line-height: 1.25em;
    }

    @media (max-width: 769px) {
      margin-top: -23px;
    }
  }
`;
