// @britishcouncil/react-solas-ors3/dist/components/auth/AuthPanelBody.js
import React, {
    useCallback
} from 'react';
import styled from '@emotion/styled';
import {
    Button
} from '@britishcouncil/react-solas';
import {
    SuiPanel
} from '../layout';
import {
    RoundIcon
} from '../icons';
import {
    LockIcon,
    KeyIcon,
    RotateCcwIcon
} from '../icons';
import {
    useNavigate
} from 'react-router-dom';
const i18nDefault = {
    subtitle: 'Sign in or register to complete your booking.',
    rebook: 'Re-book and book new tests quickly and easily',
    paySecure: 'Pay securely',
    materials: 'Access study materials',
    newUsers: 'New users',
    returningUsers: 'Returning users',
    register: 'Register',
    login: 'Log in',
};
export const AuthPanelBody = ({
    children,
    disabled,
    loading,
    loginUrl,
    changeLocation = false,
    registerUrl,
    onRedirect,
    i18n = i18nDefault,
    ...props
}) => {
    const navigate = useNavigate();
    const redirect = useCallback((targetPath) => {
        const process = async () => {
            if (onRedirect) {
                await onRedirect();
            }
            if (changeLocation) {
                window.location.href = targetPath;
                return;
            }
            navigate(targetPath);
        };
        process();
    }, [onRedirect, changeLocation, navigate]);
    return (React.createElement(SuiPanel.Body, { ...props
        },
        children,
        React.createElement(Subtitle, null, i18n.subtitle),
        React.createElement(DescriptionLine, null,
            React.createElement(RoundIcon, {
                    className: "desc-icon"
                },
                React.createElement(LockIcon, null)),
            React.createElement(LoginDescriptionText, null, i18n.paySecure)),
        React.createElement(DescriptionLine, null,
            React.createElement(RoundIcon, {
                    className: "desc-icon"
                },
                React.createElement(KeyIcon, null)),
            React.createElement(LoginDescriptionText, null, i18n.materials)),
        React.createElement(DescriptionLine, null,
            React.createElement(RoundIcon, {
                    className: "desc-icon"
                },
                React.createElement(RotateCcwIcon, null)),
            React.createElement(LoginDescriptionText, null, i18n.rebook)),
        React.createElement(CtasContainer, null,
            React.createElement(ButtonDescription, null, i18n.newUsers),
            React.createElement(Button, {
                intent: "primary",
                "data-testid": "register",
                id: "register",
                disabled: disabled,
                loading: loading,
                onClick: () => redirect(registerUrl)
            }, i18n.register),
            React.createElement(ButtonDescription, {
                style: {
                    marginTop: '1em'
                }
            }, i18n.returningUsers),
            React.createElement(Button, {
                intent: "default",
                "data-testid": "login",
                disabled: disabled,
                loading: loading,
                onClick: () => redirect(loginUrl)
            }, i18n.login))));
};
const Subtitle = styled.p `
  color: #23085a;
  font-weight: bold;
`;
const DescriptionLine = styled.div `
  margin-top: 15px;
  display: flex;
  align-items: center;
`;
const LoginDescriptionText = styled.span `
  margin-left: 15px;
  color: #000;
`;
const CtasContainer = styled.div `
  margin-top: 1.5em;
  display: flex;
  flex-direction: column;
  justify-content: stretch;

  .btn {
    margin-top: 0.25em;
  }
`;
const ButtonDescription = styled.span `
  text-align: center;
`;
