// @britishcouncil/react-extro-ui/dist/components/table/components/actions/actionsColumn.js
import React from 'react';
import {
    ActionsOpener
} from './ActionsOpener';
import {
    builtinColumns
} from '../builtinColumns';
import {
    isRowDisabled
} from '../infoRow/helpers';
/** Max number of columns for which we will show action buttons without circle opener */
const MAX_COLUMNS = 4;
export const actionsColumn = (Actions, infoRows) => {
    if (!Actions)
        return [];
    return [{
        id: builtinColumns.actions,
        header: '',
        cell: ({
            table,
            row
        }) => {
            const dataColumns = table.getAllColumns().filter((el) => !Object.values(builtinColumns).includes(el.id));
            const disabled = infoRows ? isRowDisabled(infoRows(row.original)) : false;
            return (React.createElement(ActionsOpener, {
                showOpener: dataColumns.length > MAX_COLUMNS,
                disabled: disabled
            }, Actions ? React.createElement(Actions, { ...row
            }) : null));
        },
    }, ];
};
