// @britishcouncil/react-extro-ui/dist/components/message/Message.style.js
import {
    css
} from '@emotion/react';
import styled from '@emotion/styled';
const S = {
    Container: styled.div `
    display: grid;
    grid-template-columns: min-content auto min-content;
    font-weight: 400;
    border-radius: ${({ theme }) => theme.spacing.sm};

    ${({ variant, theme }) => {
        switch (variant) {
            case 'info':
                return css `
    background: $ {
        theme.colors.segment.blue
    };
    color: $ {
        theme.colors.indigo
    };
    `;
            case 'warning':
                return css `
    background: #fff1ce;
    color: $ {
        theme.colors.brown
    };
    `;
            case 'success':
                return css `
    background: #e5faf3;
    color: $ {
        theme.colors.petrol
    };
    `;
            case 'error':
                return css `
    background: #ffdcdb;
    color: $ {
        theme.colors.red
    };
    `;
            case 'violet':
                return css `
    background: #e9dbff;
    color: $ {
        theme.colors.darkPurple
    };
    `;
            case 'grey':
            default:
                return css `
    background: $ {
        theme.colors.segment.grey
    };
    color: $ {
        theme.colors.darkGrey
    };
    `;
        }
    }};

    ${({ background }) => background && css `
    background: $ {
        background
    }
    `};

    ${({ color }) => color && css `
    color: $ {
        color
    }
    `};

    ${({ size, theme }) => {
        switch (size) {
            default:
            case 'normal':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        padding: $ {
            theme.spacing.md
        };
        font - size: $ {
            theme.fontSize.mobile.small
        };
    }

    $ {
        theme.mediaQueries.md
    } {
        grid - gap: $ {
            theme.spacing.sm
        }
        0;
        padding: $ {
            theme.spacing.md
        }
        $ {
            theme.spacing.lg
        };
        font - size: $ {
            theme.fontSize.desktop.body
        };
    }
    `;
            case 'large':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        grid - gap: $ {
            theme.spacing.sm
        }
        0;
        padding: $ {
            theme.spacing.lg
        };
        font - size: $ {
            theme.fontSize.mobile.body
        };
    }

    $ {
        theme.mediaQueries.md
    } {
        grid - gap: $ {
            theme.spacing.md
        }
        0;
        padding: $ {
            theme.spacing.xxl
        }
        $ {
            theme.spacing.xl
        };
        font - size: $ {
            theme.fontSize.desktop.body
        };
    }
    `;
        }
    }};
  `,
    Icon: styled.div `
    display: flex;
    grid-row: ${({ alignToFirstRow }) => (alignToFirstRow ? '1' : '1 / 3')};
    align-self: center;

    ${({ size, theme, hidden, iconSize }) => {
        switch (size) {
            default:
            case 'normal':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        margin - inline - end: $ {
            !hidden ? theme.spacing.md : ''
        };

        svg {
            height: $ {
                iconSize ? ? theme.spacing.md
            };
            width: $ {
                iconSize ? ? theme.spacing.md
            };
        }

        i {
            font - size: $ {
                iconSize ? ? theme.spacing.md
            };
        }
    }

    $ {
        theme.mediaQueries.md
    } {
        margin - inline - end: $ {
            !hidden ? theme.spacing.lg : ''
        };

        svg {
            height: $ {
                iconSize ? ? theme.spacing.lg
            };
            width: $ {
                iconSize ? ? theme.spacing.lg
            };
        }

        i {
            font - size: $ {
                iconSize ? ? theme.spacing.lg
            };
        }
    }
    `;
            case 'large':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        margin - inline - end: $ {
            !hidden ? theme.spacing.lg : ''
        };

        svg {
            height: $ {
                iconSize ? ? theme.spacing.lg
            };
            width: $ {
                iconSize ? ? theme.spacing.lg
            };
        }

        i {
            font - size: $ {
                iconSize ? ? theme.spacing.lg
            };
        }
    }

    $ {
        theme.mediaQueries.md
    } {
        margin - inline - end: $ {
            !hidden ? theme.spacing.xl : ''
        };

        svg {
            height: $ {
                iconSize ? ? theme.spacing.xxl
            };
            width: $ {
                iconSize ? ? theme.spacing.xxl
            };
        }

        i {
            font - size: $ {
                iconSize ? ? theme.spacing.xxl
            };
        }
    }
    `;
        }
    }};
  `,
    Header: styled.div `
    font-weight: 700;
    display: flex;
    align-items: center;

    ${({ size, theme }) => {
        switch (size) {
            default:
            case 'normal':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        font - size: $ {
            theme.fontSize.mobile.h5
        };
    }

    $ {
        theme.mediaQueries.md
    } {
        font - size: $ {
            theme.fontSize.desktop.h5
        };
    }
    `;
            case 'large':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        font - size: $ {
            theme.fontSize.mobile.h4
        };
    }

    $ {
        theme.mediaQueries.md
    } {
        font - size: $ {
            theme.fontSize.desktop.h4
        };
    }
    `;
        }
    }};
  `,
    ContentContainer: styled.div `
    grid-column-start: 2;
  `,
    SideContentContainer: styled.div `
    align-self: center;
    grid-column: 3;
    grid-row: ${({ alignToFirstRow }) => alignToFirstRow ? '1' : '1 / 3'};
    font-weight: 700;
    white-space: nowrap;

    ${({ size, theme }) => {
        switch (size) {
            default:
            case 'normal':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        margin - inline - start: $ {
            theme.spacing.md
        };
        font - size: $ {
            theme.fontSize.mobile.h6
        };
    }

    $ {
        theme.mediaQueries.md
    } {
        margin - inline - start: $ {
            theme.spacing.lg
        };
        font - size: $ {
            theme.fontSize.desktop.body
        };
    }
    `;
            case 'large':
                return css `
    $ {
        theme.mediaQueries.xs
    } {
        display: none; /* hidden when viewport for smallest device screens */
        margin - inline - start: $ {
            theme.spacing.lg
        };
        font - size: $ {
            theme.fontSize.mobile.body
        };
    }

    $ {
        theme.mediaQueries.sm
    } {
        display: block;
    }

    $ {
        theme.mediaQueries.md
    } {
        margin - inline - start: $ {
            theme.spacing.xl
        };
        font - size: $ {
            theme.fontSize.desktop.h5
        };
    }
    `;
        }
    }};
  `,
};
export default S;
