// @britishcouncil/react-solas/dist/pages/ErrorPage.js
import React from 'react';
import {
    Link
} from 'react-router-dom';
import styled from '@emotion/styled';
import {
    keyframes
} from '@emotion/react';
import {
    SimpleContainer
} from '../containers';
import {
    IcoHome,
    Icon
} from '../icons';
const StyledImgContainer = styled.div `
  margin: 0 auto 30px;
`;
const rotationAnimation = keyframes `
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;
const StyledLogo = styled.img `
  margin: 0 auto;
  animation: ${rotationAnimation} 12s linear infinite;
`;
export const GO_TO_HOME_PAGE = 'Go to home page';
/**
 * Generic error page component
 */
export function ErrorPage(props) {
    return (React.createElement(SimpleContainer, {
            centered: true,
            size: "medium"
        },
        React.createElement(StyledImgContainer, null,
            React.createElement(StyledLogo, {
                src: "https://bccdn.azureedge.net/favicons/bc-favicon.png",
                alt: "British Council Logo"
            })),
        React.createElement("aside", {
                className: "panel panel-danger"
            },
            React.createElement("h2", {
                className: "panel-heading"
            }, props.title),
            React.createElement("div", {
                    className: "panel-body"
                },
                React.createElement("p", null, props.message),
                React.createElement("div", {
                        className: "text-center"
                    },
                    React.createElement(Link, {
                            to: "/",
                            className: "btn btn-primary btn-md"
                        },
                        React.createElement(Icon, {
                                color: "white",
                                size: "tiny"
                            },
                            React.createElement(IcoHome, null)),
                        GO_TO_HOME_PAGE))))));
}
