// @britishcouncil/react-extro-ui/dist/components/table/components/checkbox/Checkbox.js
import React from 'react';
import styled from '@emotion/styled';
export const Checkbox = ({
    indeterminate,
    className = '',
    ...rest
}) => {
    const ref = React.useRef(null);
    React.useEffect(() => {
        if (typeof indeterminate === 'boolean') {
            ref.current.indeterminate = !rest.checked && indeterminate;
        }
    }, [ref, indeterminate, rest.checked]);
    return (React.createElement(Root, null,
        React.createElement("input", {
            type: "checkbox",
            ref: ref,
            ...rest
        })));
};
const Root = styled.div `
  display: flex;
  justify-content: center;
  input[type='checkbox'] {
    background: white;
    width: 16px;
    height: 16px;
    border: ${({ theme: { colors } }) => `
1 px solid $ {
    colors.darkGrey
}
`};
    border-radius: 4px;
    margin: 0 10px;

    &:checked {
      background-color: ${({ theme: { colors } }) => colors.darkBlue};
      border: 1px solid ${({ theme: { colors } }) => `
1 px solid $ {
    colors.darkBlue
}
`};
      box-shadow: none;
      &::before {
        content: '';
        width: 6px;
        height: 12px;
        display: block;
        transform: rotate(45deg);
        border-bottom: 2px solid white;
        border-inline-end: 2px solid white;
        margin-inline-start: 4px;
      }
    }

    &:disabled {
      cursor: default;
      opacity: 0.2;
      background: ${({ theme: { colors } }) => colors.grey};
    }
  }
`;
