// @britishcouncil/react-extro-ui/dist/components/table/components/checkbox/checkboxColumn.js
import React from 'react';
import ReactDOM from 'react-dom';
import {
    Checkbox
} from './Checkbox';
import {
    isRowDisabled
} from '../infoRow/helpers';
import {
    builtinColumns
} from '../builtinColumns';
export const checkboxColumn = (el, infoRows, disableCheckbox) => ({
    id: builtinColumns.checkbox,
    header: ({
        table
    }) => {
        const allRows = table.getRowModel().rows;
        const disabledRows = infoRows || disableCheckbox ?
            allRows.reduce((acc, curr) => {
                const disabledRow = infoRows ? isRowDisabled(infoRows(curr.original)) : false;
                const disabledCheckbox = disableCheckbox ? disableCheckbox(curr.original) : false;
                return disabledRow || disabledCheckbox ? { ...acc,
                    [curr.id]: false
                } : acc;
            }, {}) :
            {};
        const allRowsCheckedOrDisabled = allRows.filter((el) => {
            return !(el.getIsSelected() || disabledRows[el.id] !== undefined);
        }).length === 0;
        return el ? (ReactDOM.createPortal(React.createElement(Checkbox, {
            checked: allRowsCheckedOrDisabled,
            onChange: () => {
                table.toggleAllPageRowsSelected(!allRowsCheckedOrDisabled);
                table.setRowSelection((s) => {
                    return { ...s,
                        ...disabledRows
                    };
                });
            },
            "data-testid": "checkbox-all"
        }), el)) : (React.createElement(React.Fragment, null));
    },
    cell: ({
        row
    }) => {
        const disabledRow = infoRows ? isRowDisabled(infoRows(row.original)) : false;
        const disabledCheckbox = disableCheckbox ? disableCheckbox(row.original) : false;
        return (React.createElement(Checkbox, {
            checked: row.getIsSelected(),
            indeterminate: row.getIsSomeSelected(),
            onChange: row.getToggleSelectedHandler(),
            disabled: disabledRow || disabledCheckbox,
            "data-testid": "checkbox-single"
        }));
    },
});
