// @britishcouncil/react-extro-ui/dist/components/switch/Switch.js
import React from 'react';
import S from './Switch.style';
import classnames from 'classnames';
export const Switch = ({
    className,
    disable = false,
    hasError = false,
    label,
    labelPosition = 'right',
    onChange,
    value = false,
    ...restProps
}) => {
    return (React.createElement(S.Container, {
            tabIndex: 1,
            onKeyDown: (e) => {
                if (e.key === 'Enter')
                    onChange(!value);
            },
            className: classnames(className, {
                'has-error': hasError,
                'is-checked': value,
                'is-disabled': disable,
                'label-right': labelPosition === 'right',
                'label-top': labelPosition === 'top',
            })
        },
        labelPosition === 'top' && React.createElement(S.Label, {
            className: "label"
        }, label),
        React.createElement(S.Input, { ...restProps,
            disabled: disable,
            type: "checkbox",
            checked: value,
            defaultChecked: value,
            onChange: () => onChange(!value)
        }),
        React.createElement(S.SwitchWrapper, null,
            React.createElement(S.Switch, {
                    className: classnames('switch', {
                        'is-checked': value,
                    })
                },
                React.createElement(S.SwitchInner, null))),
        labelPosition === 'right' && React.createElement(S.Label, {
            className: "label"
        }, label)));
};
