// @britishcouncil/react-extro-ui/dist/components/switch/Switch.style.js
import styled from '@emotion/styled';
import {
    colorWithOpacity
} from '../../theme';
const S = {
    Container: styled.label `
    position: relative;
    align-items: center;
    cursor: pointer;
    width: auto;

    &:hover {
      .switch {
        border-color: ${({ theme }) => theme.colors.indigo};
      }
    }

    &:active {
      .switch {
        background: ${colorWithOpacity('darkBlue', 0.1)};
        border-color: ${({ theme }) => theme.colors.indigo};
      }
    }

    &.is-checked {
      & > div {
        border-color: ${({ theme }) => theme.colors.darkBlue};

        .switch {
          border-color: ${({ theme }) => theme.colors.darkBlue};
        }
      }
    }

    &.label-right {
      display: flex;

      .label {
        margin-inline-start: 16px;
        margin-bottom: 0;
      }
    }

    &.label-top {
      display: inline-block;

      .label {
        margin-inline-start: 0;
        margin-bottom: 16px;
      }
    }

    &.is-disabled {
      pointer-events: none;
      opacity: 0.4;

      input {
        pointer-events: auto;
      }
    }

    &.has-error {
      color: ${({ theme }) => theme.colors.darkRed};
      & > div {
        border-color: ${({ theme }) => theme.colors.darkRed};
        background: ${colorWithOpacity('darkRed', 0.1)} !important;
      }

      .switch {
        border-color: ${({ theme }) => theme.colors.darkRed} !important;
      }

      &.is-checked {
        .switch {
          span {
            background: ${({ theme }) => theme.colors.darkRed} !important;
          }
        }
      }
    }
  `,
    Input: styled.input `
    opacity: 0;
    width: 0;
    height: 0;
    display: none;

    &:checked + div {
      background: ${colorWithOpacity('darkBlue', 0.1)};

      .switch {
        transform: translateX(29px);
        border: 1px solid;
      }
    }
  `,
    SwitchWrapper: styled.div `
    width: 60px;
    height: 30px;
    background: transparent;
    border: 1px solid ${({ theme }) => theme.colors.grey};
    transition: 0.3s;
    border-radius: 30px;
    position: relative;
  `,
    Switch: styled.span `
    position: absolute;
    content: '';
    height: 25px;
    width: 25px;
    left: 3px;
    bottom: 1.6px;
    border: 1px solid ${({ theme }) => theme.colors.grey};
    border-radius: 50%;
    transition: 0.3s;
    padding: 2px;
    background: #fff;
    display: flex;
    justify-content: center;
    align-items: center;

    &.is-checked {
      span {
        background: ${({ theme }) => theme.colors.darkBlue};
      }
    }
  `,
    Label: styled.p `
    font-size: 16px;
    line-height: 30px;
    line-height: 18px;
    font-weight: 400;
  `,
    SwitchInner: styled.span `
    border-radius: 50%;
    height: 100%;
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;
  `,
};
export default S;
