// @britishcouncil/react-extro-ui/dist/theme/theme.js
export const theme = {
    colors: {
        text: {
            body: '#54565A',
            link: '#005CB9',
            header: '#23085A',
        },
        indigo: '#23085a',
        yellow: '#ffe300',
        orange: '#ff8200',
        red: '#ea0034',
        pink: '#ff00c8',
        purple: '#b25eff',
        blue: '#00dcff',
        teal: '#00edc3',
        green: '#5deb4b',
        petrol: '#035c67',
        grey: '#898a8d',
        brown: '#7e3c23',
        darkYellow: '#e4b800',
        darkOrange: '#ff4635',
        darkRed: '#ab0033',
        darkPink: '#920061',
        darkPurple: '#7716bd',
        darkBlue: '#005cb9',
        darkTeal: '#009d8c',
        darkGreen: '#00953b',
        darkPetrol: '#003b4a',
        darkGrey: '#54565a',
        darkBrown: '#472029',
        lightYellow: '#ffe49e',
        lightOrange: '#ffc288',
        lightRed: '#ffbab8',
        lightPink: '#ffa0d7',
        lightPurple: '#d3b8ff',
        lightBlue: '#97daff',
        lightTeal: '#99edd1',
        lightGreen: '#c0df88',
        lightPetrol: '#7ca7ad',
        lightGrey: '#c8c8c8',
        lightBrown: '#dcbfa5',
        segment: {
            blue: '#E7F0F7',
            grey: '#F4F4F4',
            teal: '#F0FCF8',
            white: '#FFFFFF',
        },
        dropdown: {
            background: {
                default: '#FFFFFF',
                hover: '#E8F2F7',
                selected: '#005CB9',
                active: '#00376F',
                disabled: '#f3f3f3',
            },
            text: {
                default: '#54565A',
                hover: '#000000',
                selected: '#FFFFFF',
                active: '#FFFFFF',
                disabled: '#a3a4a6',
            },
        },
        placeholder: {
            default: '#767676',
            error: '#C04066',
        }
    },
    spacing: {
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '24px',
        xl: '32px',
        xxl: '48px',
    },
    fontSize: {
        default: '16px',
        desktop: {
            h1: '48px',
            h2: '40px',
            h3: '32px',
            h4: '24px',
            h5: '20px',
            h6: '16px',
            body: '18px',
            small: '14px',
        },
        mobile: {
            h1: '28px',
            h2: '28px',
            h3: '24px',
            h4: '20px',
            h5: '18px',
            h6: '14px',
            body: '16px',
            small: '12px',
        },
    },
    breakpoints: {
        // Breakpoints based on Solas responsiveness
        xs: 0,
        sm: 576,
        md: 768,
        lg: 992,
        xl: 1200,
    },
    mediaQueries: {
        xs: `@media (min-width: ${0}px)`,
        sm: `@media (min-width: ${576}px)`,
        md: `@media (min-width: ${768}px)`,
        lg: `@media (min-width: ${992}px)`,
        xl: `@media (min-width: ${1200}px)`,
    },
    elevation: {
        0: 'box-shadow: none;',
        1: 'box-shadow: 0px 1px 4px 0px rgba(84, 86, 90, 0.24);',
        2: 'box-shadow: 0px 2px 8px 0px rgba(84, 86, 90, 0.24);',
        3: 'box-shadow: 0px 4px 16px 0px rgba(84, 86, 90, 0.24);',
        4: 'box-shadow: 0px 8px 32px 0px rgba(84, 86, 90, 0.24);',
        5: 'box-shadow: 0px 16px 64px 0px rgba(84, 86, 90, 0.24);',
    },
};
