// @britishcouncil/react-solas/dist/components/alerts/BcAlert.js
import React from 'react';
import cn from 'classnames';
import {
    Icon,
    IcoCancelCircled
} from '../..';
export const BcAlert = (props) => {
    const onClose = () => {
        props.onClose && props.onClose();
    };
    const classes = cn(`alert bc-alert-${props.type}`, props.className, {
        'alert-dismissible': props.dismissible,
    });
    return (React.createElement("div", {
            role: "alert",
            ...props,
            className: classes
        },
        props.dismissible ? React.createElement(DismissButton, {
            onClose: onClose,
            type: props.type
        }) : React.createElement(React.Fragment, null),
        props.title ? React.createElement("h2", {
            className: "alert-title"
        }, props.title) : React.createElement(React.Fragment, null),
        props.children || React.createElement(React.Fragment, null)));
};
const DismissButton = (props) => {
    return (React.createElement("button", {
            type: "button",
            className: "close",
            "data-dismiss": "alert",
            "aria-label": "Close",
            onClick: props.onClose
        },
        React.createElement(Icon, {
                size: "large",
                link: true,
                color: props.type === 'news' ? 'white' : '#8b6c3d'
            },
            React.createElement(IcoCancelCircled, null))));
};
