// @britishcouncil/react-solas/dist/components/footers/SimpleFooter.js
import React from 'react';
import {
    DateTime
} from 'luxon';
import {
    SimpleContainer
} from '../../containers';
export const SimpleFooter = ({
    accessibilityUrl,
    centered,
    dataProtectionUrl,
    privacyAndCookiesUrl,
    size,
    termsOfUseUrl,
    copyrightOwner,
    linksCustomText,
    footerCustomText,
}) => {
    accessibilityUrl = accessibilityUrl || 'https://www.britishcouncil.org/accessibility';
    dataProtectionUrl = dataProtectionUrl || 'https://www.britishcouncil.org/privacy-cookies/data-protection';
    privacyAndCookiesUrl = privacyAndCookiesUrl || 'https://www.britishcouncil.org/privacy-cookies';
    termsOfUseUrl = termsOfUseUrl || 'https://www.britishcouncil.org/terms';
    copyrightOwner = copyrightOwner || 'British Council';
    return (React.createElement("footer", {
            className: "bc-footer"
        },
        React.createElement(SimpleContainer, {
                size: size,
                centered: centered
            },
            React.createElement("ul", {
                    className: "bc-site-links"
                },
                React.createElement("li", null,
                    React.createElement("a", {
                        target: "_blank",
                        rel: "noopener noreferrer",
                        href: termsOfUseUrl
                    }, linksCustomText ? .termsOfUse ? ? `Terms of use`)),
                React.createElement("li", null,
                    React.createElement("a", {
                        target: "_blank",
                        rel: "noopener noreferrer",
                        href: accessibilityUrl
                    }, linksCustomText ? .accessibility ? ? `Accessibility`)),
                React.createElement("li", null,
                    React.createElement("a", {
                        target: "_blank",
                        rel: "noopener noreferrer",
                        href: privacyAndCookiesUrl
                    }, linksCustomText ? .privacyAndCookies ? ? `Privacy and cookies`)),
                React.createElement("li", null,
                    React.createElement("a", {
                        target: "_blank",
                        rel: "noopener noreferrer",
                        href: dataProtectionUrl
                    }, linksCustomText ? .dataProtection ? ? `Data protection`))),
            React.createElement("div", {
                    className: "bc-copyright"
                },
                React.createElement("div", null,
                    "\u00A9 ",
                    DateTime.local().year,
                    " ",
                    copyrightOwner),
                React.createElement("p", null,
                    footerCustomText ? .mainInfo ? ?
                    `The United Kingdom's international organisation for cultural relations and educational opportunities.`,
                    React.createElement("br", null),
                    footerCustomText ? .additionalInfo ? ?
                    `A registered charity: 209131 (England and Wales) SC037733 (Scotland).`)))));
};
