// @britishcouncil/react-solas-ors3/dist/components/forms/password/PasswordRequirements.js
import React from 'react';
import cn from 'classnames';
import styled from '@emotion/styled';
import {
    SimpleProgressBar
} from '@britishcouncil/react-solas';
import {
    CheckItem
} from './CheckItem';
export const PasswordRequirements = ({
    pswStrength,
    i18n = i18nDefault
}) => {
    const pswStrengthDict = {
        0: {
            className: 'danger',
            name: i18n.levels['0']
        },
        1: {
            className: 'danger',
            name: i18n.levels['1']
        },
        2: {
            className: 'warning',
            name: i18n.levels['2']
        },
        3: {
            className: 'success',
            name: i18n.levels['3']
        },
        4: {
            className: 'success',
            name: i18n.levels['4']
        },
    };
    return (React.createElement(StyledWell, {
            className: "well",
            "aria-live": "polite"
        },
        React.createElement("div", {
                className: cn('password-requirements help-block', {
                    'has-success': pswStrength.isCompliant,
                })
            },
            pswStrength.isCompliant && React.createElement("div", {
                className: "password-ok"
            }, i18n.ok), !pswStrength.isCompliant && (React.createElement("div", {
                    className: "checklist-container"
                },
                React.createElement("span", null, i18n.mustHave),
                React.createElement("ul", {
                        className: "checklist"
                    },
                    React.createElement(CheckItem, {
                        checked: pswStrength.isLongEnough,
                        text: i18n.reqs.length
                    })),
                React.createElement("span", null, i18n.shouldHave),
                React.createElement("ul", {
                        className: "checklist"
                    },
                    React.createElement(CheckItem, {
                        checked: pswStrength.hasLowercase,
                        text: i18n.reqs.lower
                    }),
                    React.createElement(CheckItem, {
                        checked: pswStrength.hasUppercase,
                        text: i18n.reqs.upper
                    }),
                    React.createElement(CheckItem, {
                        checked: pswStrength.hasNumber,
                        text: i18n.reqs.number
                    }),
                    React.createElement(CheckItem, {
                        checked: pswStrength.hasSpecialChars,
                        text: i18n.reqs.special
                    })))),
            React.createElement(SimpleProgressBar, {
                percentage: (100 * pswStrength.pswStrength) / 4,
                intent: pswStrengthDict[pswStrength.pswStrength].className
            }),
            React.createElement("p", {
                    className: "mt-2 mb-0"
                },
                React.createElement("b", null,
                    i18n.strength,
                    ": ",
                    pswStrengthDict[pswStrength.pswStrength].name)))));
};
const StyledWell = styled.div `
  padding: 10px 20px 20px;
  position: absolute;
  left: 0;
  right: 0;
  top: 100%;
  z-index: 1000;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.16);

  .password-requirements {
    margin-bottom: 0;
    position: relative;

    > div {
      transition: all ease-in-out 0.3s;
    }

    > p {
      margin: 0.5rem 0 0;
    }

    .checklist-container {
      .checklist {
        list-style: none;
        margin: 5px 0 7px;
      }
    }

    .progress {
      margin: 15px 0 0;
      height: 5px;
    }
  }
`;
const i18nDefault = {
    levels: {
        '0': 'Very Weak',
        '1': 'Weak',
        '2': 'Medium',
        '3': 'Strong',
        '4': 'Very Strong',
    },
    ok: 'Your password meets requirements',
    mustHave: 'Your password must have:',
    shouldHave: 'And it needs to meet 3 of the following requirements:',
    strength: 'Strength',
    reqs: {
        length: '8 or more characters',
        lower: 'Lowercase letters',
        upper: 'Uppercase letters',
        number: 'At least one number',
        special: 'At least one special character',
    },
};
