// @britishcouncil/react-solas/dist/components/headers/NavbarHeader.js
import React from 'react';
import styled from '@emotion/styled';
const StyledAppName = styled.span `
  font-size: 24px;
  line-height: 34px;
  margin-left: 5px;
  padding-left: 0;
  color: #fff;
`;
const StyledNavbarBrand = styled.div `
  ${props => props.useSoftMarginTop && `
    @media (min-width: 850px) {
      .navbar-brand {
        margin-top: 25px !important;
      }
    }
  `}
`;
export function NavbarHeader(props) {
    let logoStyle = {};
    if (props.logoUrl) {
        logoStyle = {
            backgroundImage: 'url(' + props.logoUrl + ')',
        };
    }
    return (React.createElement("div", {
            className: "navbar-header",
            style: props.style
        },
        React.createElement(StyledNavbarBrand, {
                useSoftMarginTop: !!props.useSoftMarginTop
            },
            React.createElement("div", {
                    className: "navbar-brand"
                },
                React.createElement("a", {
                    href: "/",
                    className: "bc-brand-logo",
                    style: logoStyle
                }, "British Council"),
                props.slogan &&
                (React.isValidElement(props.slogan) ? (props.slogan) : (React.createElement("span", {
                    className: "bc-brand-slogan"
                }, props.slogan))),
                props.additionalTitle ? React.createElement(StyledAppName, null, props.additionalTitle) : null))));
}
