// @britishcouncil/react-solas/dist/components/headers/PrimaryNav.js
import React, {
    useContext
} from 'react';
import {
    Icon,
    IcoHome
} from '../../icons';
import {
    NavLink
} from 'react-router-dom';
import {
    MobileMenuContext
} from '.';
export function PrimaryNav(props) {
    return (React.createElement("ul", {
            id: props.id,
            className: "nav navbar-nav bc-microsite-nav navbar-nav-with-bg navbar-nav-primary"
        },
        props.showHome && React.createElement(NavbarHomeLink, null),
        props.children));
}

function NavbarHomeLink() {
    const {
        closeMenu
    } = useContext(MobileMenuContext);
    return (React.createElement("li", null,
        React.createElement(NavLink, {
                end: true,
                to: "/",
                onClick: closeMenu
            },
            React.createElement("span", {
                    className: "hidden-xs hidden-sm"
                },
                React.createElement(Icon, {
                        color: "white"
                    },
                    React.createElement(IcoHome, null))),
            React.createElement("span", {
                className: "hidden-md hidden-lg"
            }, "Home"))));
}
