// @britishcouncil/react-extro-ui/dist/components/header/Header.js
import React from 'react';
import {
    Logo
} from './components/logo/Logo';
import {
    LanguageSelector
} from '../languageSelector/LanguageSelector';
import * as S from './Header.styles';
export const Header = ({
    logoUrl,
    slogan,
    translations
}) => {
    return (React.createElement(S.Root, null,
        React.createElement("div", {
                className: "container"
            },
            React.createElement("div", {
                    className: "header-container"
                },
                React.createElement("div", {
                        className: "left-section"
                    },
                    React.createElement(Logo, {
                        logoUrl: logoUrl || 'https://bccdn.britishcouncil.org/dev/bc-logo-white.svg',
                        alt: "bc-logo"
                    }),
                    slogan && (React.isValidElement(slogan) ? slogan : React.createElement("span", {
                        className: "slogan"
                    }, slogan))),
                React.createElement("div", {
                    className: "right-section"
                }, translations ? .enabled && (React.createElement(LanguageSelector, {
                    availableLanguages: translations.availableLanguages,
                    initialValue: translations.defaultLanguageCode,
                    onChange: translations.onLanguageChange
                })))))));
};
