// @britishcouncil/react-extro-ui/dist/components/radio/Radio.styles.js
import styled from '@emotion/styled';
import {
    colorWithOpacity
} from '../../theme';
export const Container = styled.label `
  display: flex;
  align-items: center;
  cursor: pointer;

  &.no-frame {
    &.error {
      .radio {
        background-color: transparent;
      }
    }

    .radio {
      height: 16px;
      width: 16px;

      border: none;
      background-color: transparent;
    }

    span {
      &:nth-child(3) {
        margin-inline-start: 8px;
      }
    }
  }

  &:hover {
    .radio {
      &:before {
        border-color: ${({ theme }) => theme.colors.indigo};
      }
    }
  }

  &:active {
    .radio {
      background-color: ${() => colorWithOpacity('darkBlue', 0.1)};
      border-color: ${({ theme }) => theme.colors.darkBlue};

      &:before {
        border-color: ${({ theme }) => theme.colors.indigo};
      }
    }
  }

  &.disabled {
    pointer-events: none;
    opacity: 0.25;
  }

  &.checked {
    .radio {
      border-color: ${({ theme }) => theme.colors.darkBlue};
      background-color: ${() => colorWithOpacity('darkBlue', 0.1)};

      &:before {
        border-color: ${({ theme }) => theme.colors.darkBlue};
      }

      &:after {
        background-color: ${({ theme }) => theme.colors.darkBlue};
      }
    }

    &.error {
      .radio {
        &:after {
          background-color: ${({ theme }) => theme.colors.darkRed};
        }
      }
    }
  }

  &.error {
    span {
      color: ${({ theme }) => theme.colors.darkRed};
    }

    .radio {
      border-color: ${({ theme }) => theme.colors.darkRed};
      background-color: ${() => colorWithOpacity('darkRed', 0.1)};

      &:before {
        border-color: ${({ theme }) => theme.colors.darkRed};
      }
    }
  }
`;
export const Radio = styled.span `
  position: relative;
  display: inline-block;
  height: 44px;
  width: 44px;
  border-radius: 50%;
  border: 1px solid ${({ theme }) => theme.colors.grey};

  &:before {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    height: 16px;
    width: 16px;
    border-radius: 50%;
    border: 1px solid ${({ theme }) => theme.colors.grey};
    background-color: #fff;
  }

  &:after {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    border-radius: 50%;
    transform: translate(-50%, -50%);
    height: 12px;
    width: 12px;
  }
`;
export const Label = styled.span `
  margin-inline-start: 16px;
  font-size: 16px;
  color: #333333;
`;
