// @britishcouncil/react-extro-ui/dist/components/multiSelect/MultiSelect.style.js
import styled from '@emotion/styled';
import {
    MultiSelect
} from 'react-multi-select-component';
import {
    colorWithOpacity
} from '../../theme';
export const MultiSelectContainer = styled(MultiSelect)
`
  &.disabled {
    .dropdown-container {
      background-color: ${({ theme }) => theme.colors.lightGrey};
      border-color: ${({ theme }) => theme.colors.lightGrey};

      .dropdown-heading-value {
        .gray {
          color: ${({ theme }) => theme.colors.darkGrey}99;
        }
      }
    }
  }
  .select-item {
    padding: 0 !important;

    &:hover,
    &:focus {
      background: linear-gradient(0deg, rgba(151, 218, 255, 0.15), rgba(151, 218, 255, 0.15)), #f7f7f7;
    }

    &:active {
      background: ${({ theme }) => theme.colors.darkBlue} !important;

      span {
        color: #fff;
      }
    }

    &.selected {
      background: ${({ theme }) => theme.colors.darkBlue} !important;

      span {
        color: #fff;
      }

      &:hover,
      &:focus {
        background: ${({ theme }) => theme.colors.darkBlue} !important;
      }

      &:active {
        background: #00376f !important;
      }
    }
  }

  .dropdown-container {
    height: 100%;
    min-height: 52px;
    border-radius: 4px;
    border: 2px solid ${({ theme }) => theme.colors.darkGrey};
    display: flex;
    align-items: center;
    padding: 8px;
  }

  .dropdown-content {
    left: 0;

    .search {
      border: none;

      input {
        height: 48px;
        border: 1px solid #e3e3e3;
        padding: 16px;
        box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.05);

        ::placeholder {
          opacity: 0.3;
        }
      }
    }
  }

  .dropdown-heading {
    height: 100% !important;
    gap: 14px;

    .dropdown-heading-value {
      border-inline-end: 1px solid ${({ theme }) => theme.colors.grey};
    }
  }

  .options {
    padding: 0;

    ::-webkit-scrollbar {
      width: 11px;
    }

    ::-webkit-scrollbar-track {
      background-color: #e5e5e5;
    }
    ::-webkit-scrollbar-thumb {
      background-color: #959a9d;
      border-radius: 9px;
    }
    ::-webkit-scrollbar-button {
      display: none;
    }
  }

  .dropdown-heading-value {
    & > span {
      display: flex;
      flex-wrap: wrap;
      gap: 4px;

      & > span {
        padding: 8px 16px;
        border-radius: 100px;
        background: ${({ theme }) => theme.colors.darkBlue};
        color: #fff;
        font-style: normal;
        font-weight: 700;
        font-size: 16px;
        line-height: 16px;
        height: 32px;
        cursor: pointer;

        svg {
          margin-inline-start: 8px;
          height: 10px;
          width: 10px;
          path {
            stroke: #fff;
          }
        }
      }
    }
  }
`;
export const ItemContainer = styled.div `
  padding: 24px 32px;
  border-bottom: 1px solid ${() => colorWithOpacity('lightGrey', 0.5)};

  &.compact {
    padding: 16px;
  }

  & > div {
    padding: 0;
  }

  .select-all-option {
    font-weight: bold;
  }
`;
