// @britishcouncil/react-solas-ors3/dist/components/errors/ErrorModal.js
import React, {
    useEffect,
    useState
} from 'react';
import AriaModal from '@justfixnyc/react-aria-modal';
import {
    Button
} from '@britishcouncil/react-solas';
import {
    SuiPanel
} from '../layout';
import {
    LinkButton
} from '../general';
import {
    HeaderContainer,
    HeaderText
} from '../auth/styles';
const i18nDefault = {
    title: 'Failure',
    header: "We're sorry",
    close: 'Close',
};
export const ErrorModal = ({
    show,
    errorMessage = 'It looks like something went wrong. Please try again in a few minutes.',
    onExit,
    onShow,
    onHide,
    i18n = i18nDefault,
}) => {
    const [prevState, setPrevState] = useState(null);
    useEffect(() => {
        if (prevState && !show && !!onHide) {
            onHide();
        }
        setPrevState(show);
    }, [show, onHide, prevState]);
    return show ? (React.createElement(AriaModal, {
            titleText: i18n.title,
            onExit: onExit,
            onEnter: onShow,
            verticallyCenter: true,
            focusDialog: true,
            underlayStyle: {
                paddingTop: '2em'
            }
        },
        React.createElement(SuiPanel, {
                intent: "danger"
            },
            React.createElement(SuiPanel.Header, null,
                React.createElement(HeaderContainer, null,
                    React.createElement(HeaderText, null, i18n.header),
                    React.createElement(LinkButton, {
                        onClick: onExit,
                        "data-testid": "close"
                    }, i18n.close))),
            React.createElement(SuiPanel.Body, null,
                React.createElement("p", null, errorMessage),
                React.createElement("div", null,
                    React.createElement(Button, {
                        onClick: onExit,
                        id: "hide-modal",
                        intent: "primary"
                    }, i18n.close)))))) : null;
};
