// @britishcouncil/react-solas-ors3/dist/components/general/BcCheckbox.js
import React from 'react';
import cn from 'classnames';
import styled from '@emotion/styled';
export const BcCheckbox = ({
    checked,
    children,
    handleChange,
    background,
    inline,
    pill,
    ...props
}) => (React.createElement(CheckBox, {
        className: cn(inline ? 'checkbox-inline' : 'checkbox', background, pill ? 'pill' : '')
    },
    React.createElement("label", null,
        React.createElement("input", {
            type: "checkbox",
            value: "0",
            ...props,
            checked: checked,
            onChange: (ev) => handleChange(ev.target.checked)
        }),
        React.createElement("span", null, children))));
const CheckBox = styled.div `
  &.pill {
    label {
      padding: 0;

      input + span {
        display: block;
        padding: 10px 20px 10px 46px;
        font-size: 0.9em;
        background: white;
        border: 1px solid white;
        border-radius: 30px;
      }

      input,
      input:checked {
        left: 18px;
        box-shadow: none !important;
      }

      input:checked + span {
        background: #f2f7fb;
        border: 1px solid #005cb9;
      }
    }
  }

  &.checbox-inline.pill label {
    min-height: 38px;
  }
`;
