// @britishcouncil/react-solas-ors3/dist/components/general/BcSelectWrapper.js
import styled from '@emotion/styled';
export const BcSelectWrapper = styled.div `
  /**
 * Main wrapper
 */
  .select-search {
    position: relative;
    box-sizing: border-box;

    *,
    *::after,
    *::before {
      box-sizing: inherit;
    }
  }

  .is-loading .select-search__indicator.using-getoptions {
    .loader {
      display: block;
    }
    .sort {
      display: none;
    }
  }

  .select-search__indicator.using-getoptions {
    .loader {
      display: none;
    }
  }

  /**
 * Value wrapper
 */
  .select-search__value {
    position: relative;
    z-index: 1;

    &::before {
      content: '';
      display: inline-block;
      position: absolute;
      top: 0;
      right: 0;
      width: 45px;
      height: 44px;
      pointer-events: none;
      background: #005cb9;
      border-top-right-radius: 3px;
      border-bottom-right-radius: 3px;

      ${({ forceLtr }) => forceLtr
    ? ''
    : ` [dir = 'rtl'] & {
    left: 0;
    right: initial;
}
`}
    }

    .select-search__indicator {
      position: absolute;
      top: 9px;
      right: 14px;
      width: 21px;
      height: 23px;
      pointer-events: none;
      color: rgba(255, 255, 255, 0.5);
      transition: color 0.15s ease-in-out;
      ${({ forceLtr }) => forceLtr
    ? ''
    : ` [dir = 'rtl'] & {
    left: 14 px;
    right: initial;
}
`}

      .loader {
        margin-top: -7px;
        margin-left: -7px;

        ${({ forceLtr }) => forceLtr
    ? ''
    : ` [dir = 'rtl'] & {
    margin - right: -7 px;
}
`}

        path {
          fill: #fff;
        }
      }
    }

    .select-search__icon {
      position: absolute;
      top: 9px;
      left: 10px;
      width: 24px;
      height: 24px;
      color: rgba(255, 255, 255, 0.5);

      ${({ forceLtr }) => forceLtr
    ? ''
    : ` [dir = 'rtl'] & {
    right: 10 px;

    &
    svg {
        transform: scaleX(1) !important;
    }
}
`}

      svg {
        transform: scaleX(1) !important;
      }
    }
  }

  /**
 * Input
 */
  .select-search__input {
    display: block;
    height: 44px;
    width: 100%;
    padding: 6px 14px;
    background: #fff;
    border: 1px solid #b3b3b3;
    border-radius: 3px;
    outline: none;
    text-align: left;
    text-overflow: ellipsis;
    line-height: 48px;
    font-size: 16px;
    -webkit-appearance: none;
    padding-left: 42px;

    &::-webkit-search-decoration,
    &::-webkit-search-cancel-button,
    &::-webkit-search-results-button,
    &::-webkit-search-results-decoration {
      -webkit-appearance: none;
    }

    &:not([readonly]):focus {
      cursor: initial;
    }
    &:focus {
      border-color: #005cb9;
      box-shadow: inset 0 0 0 1px #005cb9;
    }

    ${({ forceLtr }) => forceLtr
    ? ''
    : ` [dir = 'rtl'] & {
    text - align: right;
    padding - right: 50 px;
}
`}
  }

  /**
 * Options wrapper
 */
  .select-search__select {
    background: #fff;
    box-shadow: 0 0.0625rem 0.125rem rgba(0, 0, 0, 0.3);
    margin-top: 3px;
  }

  /**
 * Options
 */
  .select-search__options {
    list-style: none;
    padding: 0;
    margin: 0;

    > li {
      padding: 0;
      margin: 0;
    }
  }

  /**
 * Option row
 */
  .select-search__row:not(:first-of-type) {
    border-top: 1px solid #eee;
  }

  /**
 * Option
 */
  .select-search__option,
  .select-search__not-found {
    display: block;
    min-height: 36px;
    width: 100%;
    padding: 5px 16px;
    background: #fff;
    border: none;
    outline: none;
    font-size: 16px;
    text-align: left;
    cursor: pointer;
    line-height: 18px;

    ${({ forceLtr }) => forceLtr
    ? ''
    : ` [dir = 'rtl'] & {
    text - align: right;
}
`}

    svg {
      fill: #54565a;
    }

    &.is-selected {
      background: #005cb9;
      color: #fff;

      svg {
        fill: #fff;
      }
    }

    &.is-highlighted,
    &:not(.is-selected):not(.select-search__not-found):hover {
      background: #ecf5ff;
    }

    &.is-highlighted.is-selected,
    &.is-selected:hover {
      background: #004fa0;
    }
  }

  .select-search--multiple .select-search__option {
    height: 48px;
  }

  .select-search__not-found {
    padding: 16px !important;
    text-align: center;
    color: #888;

    &:hover {
      cursor: default;
    }
  }

  /**
 * Group
 */
  .select-search__group-header {
    font-size: 10px;
    text-transform: uppercase;
    background: #eee;
    padding: 8px 16px;
  }

  /**
 * States
 */
  .select-search.is-disabled {
    opacity: 0.5;
  }

  .select-search:not(.is-disabled) .select-search__input {
    cursor: pointer;
  }

  /**
 * Modifiers
 */
  .select-search--multiple {
    border-radius: 0;
    overflow: hidden;

    .select-search__input {
      cursor: initial;
      border-radius: 3px;
    }

    .select-search__select {
      position: relative;
      overflow: auto;
      max-height: 260px;
      border-top: 1px solid #eee;
      border-radius: 0;
    }
  }

  .select-search--multiple:not(.select-search--search) .select-search__input {
    cursor: default;
  }

  .select-search:not(.select-search--multiple):not(.has-focus) .select-search__input:hover {
    border-color: #9a9a9a;
  }

  .select-search:not(.select-search--multiple) .select-search__select {
    position: absolute;
    z-index: 11;
    right: 0;
    left: 0;
    border-radius: 3px;
    overflow: auto;
    max-height: 300px;
  }

  .select-search:hover .select-search__indicator {
    color: rgba(255, 255, 255, 0.9);
  }
`;
