// @britishcouncil/react-solas-ors3/dist/components/general/PageContainer.js
import React from 'react';
import styled from '@emotion/styled';
import {
    LoadingContainer
} from '@britishcouncil/react-solas';
import {
    MainContainer
} from '../layout';
const StyledContainer = styled(MainContainer)
`
  background-color: ${({ transparent }) => (transparent ? 'transparent' : '#e5eff8')};
  padding: 32px 163px;

  @media (min-width: 850px) {
    max-width: 958px; /* 632px + 2 * 163px */
  }
  @media (max-width: 849px) {
    padding: 24px;
    margin: 0;
  }
`;
export const PageContainer = ({
    children,
    loading,
    transparent,
}) => (React.createElement(PageSectionRow, null,
    React.createElement(LoadingContainer, {
            loading: loading
        },
        React.createElement(StyledContainer, {
            transparent: transparent,
            "data-testid": "page-container"
        }, children))));
const PageSectionRow = styled.div `
  position: relative;
  padding-top: 10px;
  padding-bottom: 30px;

  /** Fixes temporary issue with loading container adding its own position relative */
  & > div {
    position: initial;
  }

  @media (max-width: 849px) {
    && {
      padding: 0;
      margin: 0 0 24px;
    }
  }
`;
