// @britishcouncil/react-solas-ors3/dist/components/general/Quote.js
import React from 'react';
import styled from '@emotion/styled';
import cn from 'classnames';
export const Quote = ({
    title,
    intent = 'default',
    children,
    ...props
}) => {
    return (React.createElement(StyledQuote, { ...props,
            className: cn('bc-quote', 'bc-quote-' + intent)
        },
        React.createElement("b", null, title),
        React.createElement("div", null, children)));
};
const StyledQuote = styled.div `
  position: relative;
  padding-left: 16px;
  margin-bottom: 20px;
  font-size: 16px;

  [dir='rtl'] & {
    padding-left: 0;
    padding-right: 16px;
  }

  &:before {
    content: ' ';
    position: absolute;
    top: 2px;
    bottom: 2px;
    left: 0;
    width: 5px;
    background-color: transparent;
    border-radius: 5px;

    [dir='rtl'] & {
      left: initial;
      right: 0;
    }
  }

  &.bc-quote-default {
    &:before {
      background-color: #005cb9;
    }
  }
  &.bc-quote-success {
    &:before {
      background-color: #3d9026;
    }
  }
  &.bc-quote-info {
    &:before {
      background-color: #5bc0de;
    }
  }
  &.bc-quote-warning {
    &:before {
      background-color: #ff7a29;
    }
  }
  &.bc-quote-danger {
    &:before {
      background-color: #cb2210;
    }
  }
  &.bc-quote-gray {
    &:before {
      background-color: #62787f;
    }
  }
  > b {
    display: block;
    margin-bottom: 10px;
  }
`;
