// @britishcouncil/react-solas/dist/components/loaders/DualRingLoader.js
import React from 'react';
import styled from '@emotion/styled';
import {
    keyframes
} from '@emotion/react';
/**
 * Use this loader if your app is using base loading solution.
 * It means that there is already loader in the `index.html` page.
 * It does not have any styles
 */
export function DualRingLoader(props) {
    const ClassName = props.ClassName || '';
    return (React.createElement("div", {
            className: "loader-container",
            "aria-label": "Loading contents... Please wait"
        },
        React.createElement(DualRing, {
            className: ClassName
        })));
}
const rotationAnimation = keyframes `
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;
const DualRing = styled.div `
  display: inline-block;
  width: 64px;
  height: 64px;
  &:after {
    content: ' ';
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid #cef;
    border-color: #cef transparent #cef transparent;
    animation: ${rotationAnimation} 1.2s linear infinite;
  }
`;
