// @britishcouncil/react-solas/dist/components/loaders/LoadingContainer.js
import React from 'react';
import {
    keyframes
} from '@emotion/react';
import styled from '@emotion/styled';
const spinKeyframes = keyframes({
    '0%': {
        transform: 'rotate(0deg)',
    },
    '100%': {
        transform: 'rotate(360deg)',
    },
});
const StyledContainer = styled.div `
  position: relative;

  .loading::before,
  .loading::after {
    content: ' ';
    position: absolute;
    top: 0;
    left: -30px;
    right: -30px;
    bottom: 0;
    z-index: 10;
  }

  .loading::before {
    background-color: white;
    opacity: 0.5;
    transition: all 0.1s ease-in-out;
  }

  .loading::after {
    top: calc(50% - (36px / 2));
    left: calc(50% - (36px / 2));
    border-radius: 50%;
    width: 36px;
    height: 36px;
    border: 6px solid rgba(0, 0, 0, 0.2);
    border-top-color: #000;
    animation: ${spinKeyframes} 1s infinite linear;

    [dir='rtl'] & {
      right: initial;
    }
  }

  .loading-spinner-fixed {
    &::after {
      position: fixed;
    }
  }

  @media screen and (max-width: 848px) {
    .loading::before {
      left: 0;
      right: 0;
    }
  }
`;
export const LoadingContainer = ({
    loading,
    children
}) => {
    return (React.createElement(StyledContainer, null,
        loading ? React.createElement("div", {
            className: "loading",
            "aria-label": "Loading contents... Please wait"
        }) : null,
        children));
};
