// @britishcouncil/react-solas-ors3/dist/components/forms/Field.js
import React from 'react';
import cn from 'classnames';
import {
    FieldError,
    FieldWarning
} from './';
export function getFieldProps(props) {
    const {
        error,
        label,
        touched,
        ...rest
    } = props;
    return [{
        error,
        label,
        touched
    }, rest];
}
export function Field({
    children,
    label,
    id,
    sublabel,
    touched,
    error,
    warningMessage,
    showColon = true,
}) {
    return (React.createElement("div", {
            className: cn('form-group', {
                'has-error': touched && error
            })
        },
        label ? (React.createElement("label", {
                htmlFor: id
            },
            label,
            showColon && ':')) : null,
        sublabel && React.createElement("div", {
            className: "small"
        }, sublabel),
        React.createElement("div", null, children),
        React.createElement(FieldError, {
            error: error,
            touched: touched
        }), !error && React.createElement(FieldWarning, {
            warning: warningMessage
        })));
}
