// @britishcouncil/react-solas-ors3/dist/components/forms/LastNameField.js
import React, {
    useState
} from 'react';
import {
    TextInput
} from '@britishcouncil/react-forms';
import styled from '@emotion/styled';
import {
    Field,
    getFieldProps
} from './';
export const LastNameField = ({
    label = 'Surname / family name',
    noLastNameLabel = "I don't have a surname / family name",
    noSurnameValue = '-',
    dataTestId = 'last-name',
    inputDir = undefined,
    ...props
}) => {
    const [fieldProps, inputProps] = getFieldProps(props);
    const [hasLastName, setHasLastName] = useState(inputProps.value !== '-');
    const toggleNoSurname = () => {
        setHasLastName((v) => !v);
        inputProps.onChange(hasLastName ? noSurnameValue : '');
    };
    return (React.createElement(Field, { ...fieldProps,
            label: label,
            showColon: props.showColon,
            id: inputProps ? .id
        },
        props.subnote ? React.createElement("span", {
            className: "small"
        }, props.subnote) : null,
        React.createElement(LastNameGroup, null,
            React.createElement(TextInput, {
                dir: inputDir,
                className: "form-control",
                ...inputProps,
                disabled: !hasLastName,
                "data-testid": dataTestId
            }),
            React.createElement("div", {
                    className: "checkbox-inline"
                },
                React.createElement("label", null,
                    React.createElement("input", {
                        type: "checkbox",
                        checked: !hasLastName,
                        onChange: toggleNoSurname
                    }),
                    noLastNameLabel)))));
};
export const LastNameGroup = styled.div `
  display: flex;
  .checkbox-inline {
    width: 365px;
    margin-left: 20px;

    [dir='rtl'] & {
      margin-left: 0;
      margin-right: 20px;
    }

    label {
      color: #54565a;
      padding: 7px 0 0 55px;
    }
  }

  @media (max-width: 567px) {
    flex-wrap: wrap;
    .checkbox-inline {
      margin: 10px 0;
      width: auto;
      label {
        padding: 15px 0 0 55px;
      }
    }
  }
`;
