// @britishcouncil/react-solas-ors3/dist/components/forms/RadioGroup.js
import React from 'react';
import cn from 'classnames';
import {
    Radio
} from '@britishcouncil/react-forms';
import styled from '@emotion/styled';
import {
    FieldError
} from './';
export const RadioGroup = ({
    touched,
    error,
    radios,
    disabled,
    ...radioProps
}) => {
    const hasError = touched && error;
    return (React.createElement("div", {
            className: cn('form-group', {
                'has-error': hasError,
            })
        },
        radios.map(({
            checkValue,
            label,
            testId,
            subtitle
        }) => (React.createElement("div", {
                key: checkValue,
                className: "radio"
            },
            React.createElement("label", null,
                React.createElement(Radio, {
                    checkValue: checkValue,
                    disabled: disabled,
                    "data-testid": testId,
                    ...radioProps
                }),
                label, !!subtitle && React.createElement(SubtleText, null, subtitle))))),
        hasError ? React.createElement(FieldError, {
            error: error,
            touched: touched
        }) : null));
};
const SubtleText = styled.div `
  font-size: 0.8em;
  line-height: 1.2em;
`;
