// @britishcouncil/react-solas-ors3/dist/components/forms/SelectField.js
import React from 'react';
import {
    useId
} from '@britishcouncil/react-common';
import {
    Field,
    getFieldProps
} from '.';
const i18nDefault = {
    choose: 'Choose',
};
export const SelectField = ({
    i18n = i18nDefault,
    ...props
}) => {
    const [fieldProps, {
        name,
        value,
        onChange,
        onBlur,
        options,
        disabled
    }] = getFieldProps(props);
    const generatedId = useId(`${props.name}-`);
    const id = props.id || generatedId;
    return (React.createElement(Field, { ...fieldProps,
            id: id,
            showColon: props.showColon
        },
        React.createElement("select", {
                className: "form-control",
                disabled: disabled,
                name: name,
                id: id,
                value: value,
                onChange: (e) => onChange(e.target.value),
                onBlur: (e) => onBlur()
            },
            React.createElement("option", {
                value: ""
            }, i18n.choose),
            options.map((e) => (React.createElement("option", {
                key: e.id,
                label: e.name,
                value: e.id
            }, e.name))))));
};
