// @britishcouncil/react-solas-ors3/dist/components/forms/TextField.js
import React, {
    useMemo
} from 'react';
import {
    TextInput
} from '@britishcouncil/react-forms';
import {
    useId
} from '@britishcouncil/react-common';
import {
    Field,
    getFieldProps
} from './';
const matchEverything = /.*/;
export const TextField = ({
    inputDir = undefined,
    extraMsg,
    showColon,
    ...props
}) => {
    const generatedId = useId(`${props.name}-`);
    const [fieldProps, inputProps] = getFieldProps(props);
    const {
        warning,
        value
    } = props;
    const id = props.id || generatedId;
    const warningRegex = warning ? .regex ? ? matchEverything;
    const warningVisible = useMemo(() => warningRegex ? .test(value), [warningRegex, value]);
    const warningMessage = warningVisible ? warning ? .message : undefined;
    return (React.createElement(Field, { ...fieldProps,
            id: id,
            warningMessage: warningMessage,
            showColon: showColon
        },
        props.subnote ? React.createElement("span", {
            className: "small"
        }, props.subnote) : null,
        React.createElement(TextInput, {
            dir: inputDir,
            id: id,
            className: "form-control",
            ...inputProps
        }),
        extraMsg));
};
