// @britishcouncil/react-solas-ors3/dist/components/forms/TncCheckbox.js
import React from 'react';
import cn from 'classnames';
import {
    useId
} from '@britishcouncil/react-common';
import {
    Checkbox
} from '@britishcouncil/react-forms';
import {
    FieldError
} from './';
export const TnCCheckbox = ({
    error,
    touched,
    children,
    ...inputProps
}) => {
    const generatedId = useId(`${inputProps.name}-`);
    const id = inputProps.id || generatedId;
    return (React.createElement("div", {
            className: cn('form-group form-checkbox', {
                'has-error': touched && error,
            })
        },
        React.createElement("div", {
                className: "checkbox"
            },
            React.createElement("label", {
                    htmlFor: id
                },
                React.createElement(Checkbox, { ...inputProps,
                    id: id,
                    "data-testid": "tnc"
                }),
                children)),
        touched && error && React.createElement(FieldError, {
            error: error,
            touched: touched
        })));
};
