// @britishcouncil/react-extro-ui/dist/components/table/Table.styles.js
import styled from '@emotion/styled';
import {
    spacingFn
} from '../../theme';
export const Root = styled.div `
  max-width: 1360px;
`;
export const TopBar = styled.div `
  display: flex;
  flex-wrap: wrap;
  background: #f4f4f4;
  border-radius: 12px 12px 0 0;
  margin: ${({ theme: { spacing } }) => `
$ {
    spacing.sm
}
0 `};
  min-height: 82px;
  align-items: center;
  justify-content: space-between;
`;
export const Actions = styled.div `
  display: flex;
  align-items: center;
  gap: 20px;
  &:first-of-type {
    margin-inline-start: 38px;
  }

  span.checkbox_selected {
    font-size: 14px;
    margin-inline-start: 30px;
    strong {
      font-weight: 700;
    }
  }
`;
export const ShowAll = styled.div `
  position: relative;
  padding: 16px;
  cursor: pointer;
`;
export const ShowAllButton = styled.button `
  font-size: 14px;
  font-weight: 700;
  line-height: 21px;
  color: #005cb9;
  position: relative;
  padding: 16px;
  cursor: pointer;
`;
export const Sorting = styled.div `
  position: relative;
  margin-inline-end: 16px;
  & > div {
    min-width: 230px;
  }
`;
export const StyledTable = styled.div `
  position: relative;
  padding: ${({ nested }) => (nested ? 0 : spacingFn(2))};
  background: #f4f4f4;
  border-radius: ${({ nested }) => (nested ? 0 : '12px')};

  table {
    width: 100%;
    overflow: hidden;
  }

  /* Don't nest first expanded row */
  table tr.row-extended tr.row-extended td:first-of-type {
    left: 9px;
  }

  table thead {
    font-weight: 700;
    font-size: 14px;
    color: ${({ theme, nested }) => (nested ? theme.colors.indigo : 'white')};
    tr {
      height: 36px;
      background: ${({ theme, nested }) => (nested ? '#efefef' : theme.colors.indigo)};
      th {
        padding: 10px 0;
        &:first-of-type {
          border-radius: 8px 0px 0px 0px;
        }
        &:last-of-type {
          border-radius: 0px 8px 0px 0px;
        }
      }
    }
    th .empty-column {
      padding: 0px 1rem;
    }
    th:first-of-type:empty {
      /* Keep all the time the same width of checkbox column */
      width: 80px;
    }
    th:last-of-type:empty {
      /* Keep the width of actions as small as possible to make room for other columns */
      width: 1px;
    }
  }

  tbody tr {
    position: relative;
    background: white;
    height: 72px;
    box-shadow: inset 0px -1px 0px #e2e8f0;
    &:hover {
      background: #eff9ff;
    }
    &.row-disabled {
      filter: grayscale(1);
      pointer-events: none;
      opacity: 0.6;
    }
    &.row-selected {
      background-color: #fbf8ff;
    }
    &.row-extended {
      thead {
        tr {
          &:hover {
            background: #efefef;
          }
        }
      }
      & > td:first-of-type {
        &::before {
          content: '';
          position: absolute;
          width: 6px;
          height: 100%;
          top: 0;
          left: 0;
          z-index: 2;
          background: ${({ theme }) => theme.colors.indigo};
        }
      }
    }
    &.row-space {
      height: 16px;
      &:first-of-type,
      &:last-of-type {
        display: none;
      }
    }
    td {
      position: relative;
      vertical-align: middle;
      font-size: 14px;
    }
  }
`;
