// @britishcouncil/react-extro-ui/dist/components/footer/Footer.styles.js
import styled from '@emotion/styled';
export const Root = styled.footer `
  background-color: ${({ color, theme }) => theme.colors[color]};
  padding: ${({ theme }) => theme.spacing.xl} 0;

  .container {
    margin-inline-start: unset;
    margin-inline-end: unset;
  }

  ${(props) => props.theme.mediaQueries.lg} {
    .container {
      margin-inline-start: auto;
      margin-inline-end: auto;
    }
  }
`;
export const LinksWrapper = styled.div `
  ${(props) => props.theme.mediaQueries.lg} {
    margin-top: ${({ theme }) => theme.spacing.md};

    > :not(:last-of-type) {
      border-inline-end: 1px solid ${(props) => props.theme.colors.lightGrey};
    }

    > :not(:first-of-type) {
      a {
        padding-inline-start: ${({ theme }) => theme.spacing.lg};
      }
    }
  }
`;
export const Link = styled.a `
  text-decoration: none;

  :hover {
    color: #fff;
  }

  color: #fff;
  display: inline-block;
  padding: ${({ theme }) => theme.spacing.sm} 0;
  font-size: ${({ theme }) => theme.fontSize.mobile.small};

  ${(props) => props.theme.mediaQueries.lg} {
    font-size: ${({ theme }) => theme.fontSize.default};
    padding-top: ${({ theme }) => theme.spacing.xs};
    padding-bottom: ${({ theme }) => theme.spacing.xs};
    padding-inline-end: ${({ theme }) => theme.spacing.xl};
  }
`;
export const Copyright = styled.div `
  color: #fff;
  margin-top: ${({ theme }) => theme.spacing.lg};
  font-size: ${({ theme }) => theme.fontSize.mobile.small};

  ${(props) => props.theme.mediaQueries.lg} {
    margin-top: ${({ theme }) => theme.spacing.xl};
    margin-bottom: ${({ theme }) => theme.spacing.md};
    font-size: ${({ theme }) => theme.fontSize.default};
  }
`;
