// @britishcouncil/react-extro-ui/dist/components/footer/Footer.js
import React from 'react';
import {
    DateTime
} from 'luxon';
import * as S from './Footer.styles';
const defaultValues = {
    linkCustomTexts: {
        accessibility: 'Accessibility',
        dataProtection: 'Data protection',
        privacyAndCookies: 'Privacy and cookies',
        termsOfUse: 'Terms of use',
    },
    copyrightOwner: 'British Council',
    linkUrls: {
        accessibility: 'https://www.britishcouncil.org/accessibility',
        dataProtection: 'https://www.britishcouncil.org/privacy-cookies/data-protection',
        privacyAndCookies: 'https://www.britishcouncil.org/privacy-cookies',
        termsOfUse: 'https://www.britishcouncil.org/terms',
    },
    footerCustomText: {
        additionalInfo: 'A registered charity: 209131 (England and Wales) SC037733 (Scotland).',
        mainInfo: "The United Kingdom's international organisation for cultural relations and educational opportunities.",
    },
    color: 'indigo',
};
export const Footer = ({
    linkUrls,
    linkCustomTexts,
    footerCustomText,
    copyrightOwner = defaultValues.copyrightOwner,
    color = defaultValues.color,
}) => {
    return (React.createElement(S.Root, {
            color: color
        },
        React.createElement("div", {
                className: "container"
            },
            React.createElement(S.LinksWrapper, {
                    className: "row"
                },
                React.createElement("div", {
                        className: "col-12 col-lg-auto"
                    },
                    React.createElement(S.Link, {
                        href: linkUrls ? .termsOfUse || defaultValues.linkUrls.termsOfUse
                    }, linkCustomTexts ? .termsOfUse || defaultValues.linkCustomTexts.termsOfUse)),
                React.createElement("div", {
                        className: "col-12 col-lg-auto"
                    },
                    React.createElement(S.Link, {
                        href: linkUrls ? .accessibility || defaultValues.linkUrls.accessibility
                    }, linkCustomTexts ? .accessibility || defaultValues.linkCustomTexts.accessibility)),
                React.createElement("div", {
                        className: "col-12 col-lg-auto"
                    },
                    React.createElement(S.Link, {
                        href: linkUrls ? .privacyAndCookies || defaultValues.linkUrls.privacyAndCookies
                    }, linkCustomTexts ? .privacyAndCookies || defaultValues.linkCustomTexts.privacyAndCookies)),
                React.createElement("div", {
                        className: "col-12 col-lg-auto"
                    },
                    React.createElement(S.Link, {
                        href: linkUrls ? .dataProtection || defaultValues.linkUrls.dataProtection
                    }, linkCustomTexts ? .dataProtection || defaultValues.linkCustomTexts.dataProtection))),
            React.createElement(S.Copyright, {
                    className: "row"
                },
                React.createElement("div", {
                        className: "col-12"
                    },
                    "\u00A9 ",
                    DateTime.local().year,
                    " ",
                    copyrightOwner || defaultValues.copyrightOwner,
                    React.createElement("br", null),
                    footerCustomText ? .mainInfo || defaultValues.footerCustomText.mainInfo,
                    React.createElement("br", null),
                    footerCustomText ? .additionalInfo || defaultValues.footerCustomText.additionalInfo)))));
};
