// @britishcouncil/react-solas-ors3/dist/components/layout/Footer.js
import React from 'react';
import {
    DateTime
} from 'luxon';
import cn from 'classnames';
import {
    SimpleContainer
} from '@britishcouncil/react-solas';
import styled from '@emotion/styled';
import {
    ExternalLink
} from '../general';
const BcFooter = styled.footer `
  @media (max-width: 768px) {
    padding: 15px 0;
  }
`;
const i18nDefault = {
    termsOfUse: 'Terms of use',
    accessibility: 'Accessibility',
    privacyAndCookies: 'Privacy and cookies',
    dataProtection: 'Data Protection',
    copyrightLine1: "The United Kingdom's international organisation for cultural relations and educational opportunities.",
    copyrightLine2: 'A registered charity: 209131 (England and Wales) SC037733 (Scotland).',
};
export const Footer = ({
    slim,
    accessibilityUrl,
    centered,
    dataProtectionUrl,
    privacyAndCookiesUrl,
    size,
    termsOfUseUrl,
    copyrightOwner,
    i18n = i18nDefault,
    className,
    ...props
}) => {
    accessibilityUrl = accessibilityUrl || 'https://www.britishcouncil.org/accessibility';
    dataProtectionUrl = dataProtectionUrl || 'https://www.britishcouncil.org/privacy-cookies/data-protection';
    privacyAndCookiesUrl = privacyAndCookiesUrl || 'https://www.britishcouncil.org/privacy-cookies';
    termsOfUseUrl = termsOfUseUrl || 'https://www.britishcouncil.org/terms';
    copyrightOwner = copyrightOwner || 'British Council';
    return (React.createElement(BcFooter, { ...props,
            className: cn('bc-footer', {
                'slim-footer': slim
            }, className)
        },
        React.createElement(SimpleContainer, {
                size: size,
                centered: centered
            },
            React.createElement("ul", {
                    className: "bc-site-links"
                },
                React.createElement("li", null,
                    React.createElement(ExternalLink, {
                        showIcon: false,
                        href: termsOfUseUrl,
                        "data-testid": "terms-of-use-link"
                    }, i18n.termsOfUse)),
                React.createElement("li", null,
                    React.createElement(ExternalLink, {
                        showIcon: false,
                        href: accessibilityUrl,
                        "data-testid": "accessibility-link"
                    }, i18n.accessibility)),
                React.createElement("li", null,
                    React.createElement(ExternalLink, {
                        showIcon: false,
                        href: privacyAndCookiesUrl,
                        "data-testid": "privacy-and-cookies-link"
                    }, i18n.privacyAndCookies)),
                React.createElement("li", null,
                    React.createElement(ExternalLink, {
                        showIcon: false,
                        href: dataProtectionUrl,
                        "data-testid": "data-protection-link"
                    }, i18n.dataProtection))),
            React.createElement("div", {
                    className: "bc-copyright"
                },
                React.createElement("div", null,
                    "\u00A9 ",
                    DateTime.local().year,
                    " ",
                    copyrightOwner),
                React.createElement("p", {
                        "data-testid": "copyright-paragraf"
                    },
                    i18n.copyrightLine1,
                    React.createElement("br", null),
                    i18n.copyrightLine2)))));
};
